src/frontend/js/widgets/SyllabusCourseRunsList/components/CourseProductItem/index.tsx

```diff
@@ -69,57 +76,63 @@ const CourseProductItem = ({ productId, courseCode }: Props) => {
 
   return (
     <CourseProductProvider courseCode={courseCode} productId={productId}>
-      <section className={['product-widget', hasError && 'product-widget--has-error'].join(' ')}>
-        {isFetching && (
-          <div className="product-widget__overlay">
-            <Spinner aria-labelledby="loading-course" theme="light" size="large">
-              <span id="loading-course">
-                <FormattedMessage {...messages.loading} />
-              </span>
-            </Spinner>
-          </div>
-        )}
-        {hasError && (
-          <p className="product-widget__content">
-            <Icon name={IconTypeEnum.WARNING} size="small" />
-            {productQuery.states.error}
-          </p>
-        )}
-        {!hasError && product && (
-          <>
-            <header className="product-widget__header">
-              <h3 className="product-widget__title">{product.title}</h3>
-              <strong className="product-widget__price h6">
-                {order && <FormattedMessage {...messages.purchased} />}
-                {hasPurchased && !order && <FormattedMessage {...messages.pending} />}
-                {!hasPurchased && (
-                  <FormattedNumber
-                    currency={product.price_currency}
-                    value={product.price}
-                    style="currency"
+      <OrderProvider order={order}>
+        <section className={['product-widget', hasError && 'product-widget--has-error'].join(' ')}>
```

Comment:
you could use `classnames`
```js
import c from 'classnames'
// ...
className={c('product-widget', {'product-widget--has-error': hasError})}
```

Reply:
Oh cool, I didn't know there were classnames on Richie! Even if it was already like that before I'm still changing :)